src/combinations.rs

```diff
@@ -135,6 +135,11 @@ impl<I> Iterator for Combinations<I>
         // Create result vector based on the indices
         Some(self.indices.iter().map(|i| self.pool[*i].clone()).collect())
     }
+
+    fn count(mut self) -> usize {
+        self.pool.fill();
```

Comment:
Do we really need to do `self.pool.fill`? Couldn't we just `let n = self.pool.len()+self.pool.iter.count()` (avoiding copying of elements) and then call `self.remaining_for(n)`?

As a consequence: Do we need `LazyBuffer::fill`?

Reply:
Couldn't `remaining_for` also be a free function (possibly accepting more arguments as required)?

Reply:
Well that's a possibility: `fn remaining_for(n: usize, k: usize, first: bool, indices: &[usize]) -> Option<usize>`
I don't think it would be elegant but we could discard `fill` I guess. I don't think it's a big problem to fill the buffer because if we did generate a few combinations (`n-k+1`, really small compared to the number of combinations in most cases), it would be filled anyway. But it's possible.

Reply:
Done in https://github.com/rust-itertools/itertools/pull/729/commits/2ab0ff199293b68f1e8ff38db373c439dac24222